index.js

```diff
@@ -11,28 +11,150 @@
 //As our previous Battleship, the winner is the player that hits the 4 opponent's ships first
 //one more Thing create a 'reset' and a 'new game' buttons as childs of the element with the id 'buttons'. the reset button has to start the game again and the new game create a new game with new players and a new random board.
 
+// declare variables and access ids in the DOM
 const board_Player1 = document.getElementById('board_player1');
+const board_Player2 = document.getElementById('board_player2');
 
-for (var x = 0; x < 4; x++) {
+const name_Player1 = document.getElementById('name_player1');
+const name_Player2 = document.getElementById('name_player2');
 
+name_Player1.textContent = prompt('Enter name for Player 1');
+name_Player2.textContent = prompt('Enter name for Player 2');
+
+const player1_lives = document.getElementById('ships_player1');
+const player2_lives = document.getElementById('ships_player2');
+
+const playerTurn = document.getElementById('turn_player');
+
+// store data of each player in two Player objects
+const player1 = {
+  name: name_Player1.textContent,
+  boatsLeft: player1_lives.innerHTML
+};
+
+const player2 = {
+  name: name_Player2.textContent,
+  boatsLeft: player2_lives.innerHTML
+};
+
+player1_lives.textContent = 4;
+player2_lives.textContent = 4;
+
+createBoard(board_Player1);
+addShips(board_Player1);
+createBoard(board_Player2);
+addShips(board_Player2);
+
+
+// create new and reset buttons
+const buttons = document.getElementById('buttons');
+const resetButton = document.createElement('button');
+const newButton = document.createElement('button');
+buttons.appendChild(resetButton);
+buttons.appendChild(newButton);
+resetButton.textContent = "Reset";
+newButton.textContent = "New";
+
+resetButton.addEventListener('click', bothButtons);
+newButton.addEventListener('click', bothButtons);
+
+// create function to new and reset buttons
+function  bothButtons (event) {
+  let button = event.target;
+  if (button === newButton) {
+    location.reload();
+  } else if (button === resetButton) {
+    player1_lives.innerText = 4;
+    player2_lives.innerText = 4;
+    board_Player1.innerText = "";
+    board_Player2.innerText = "";
+    createBoard(board_Player1);
+    addShips(board_Player1);
+    createBoard(board_Player2);
+    addShips(board_Player2);
+  }
+}
+
+
+// create function that adds ships randomly to the DOM
+function addShips(board) {
+  let ship = 0;
+  while (ship < 4) {
+    let xPosition = Math.floor(Math.random() * 4);
+    let yPosition = Math.floor(Math.random() * 4);
+    let cell1 = board.getElementsByTagName('li')[xPosition];
+    let cell2 = cell1.getElementsByTagName('div')[yPosition];
+      if (cell2.value === 0) {
+        cell2.value = 1;
+        ship++;
+      } else {
+        continue;
+      }
+  }
+}
+
+// function that create board cells
+function createBoard(cells) {
+  for (var x = 0; x < 4; x++) {
```

Comment:
Using `let` rather than `var` in all the for loops would be nicer and more reliable.